src/Microsoft.Diagnostics.Monitoring.StartupHook/Exceptions/Pipeline/Steps/FilterMonitorExceptionPipelineStep.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.Diagnostics.Monitoring.StartupHook.Exceptions.Pipeline.Steps
+{
+    /// <summary>
+    /// An exception pipeline step that will only allow an exception to be
+    /// processed further if it was not caused by a dotnet-monitor in-proc feature.
+    /// </summary>
+    internal sealed class FilterMonitorExceptionPipelineStep
+    {
+        private readonly ExceptionPipelineDelegate _next;
+
+        public FilterMonitorExceptionPipelineStep(ExceptionPipelineDelegate next)
+        {
+            ArgumentNullException.ThrowIfNull(next);
+
+            _next = next;
+        }
+
+        public void Invoke(Exception exception, ExceptionPipelineExceptionContext context)
+        {
+            ArgumentNullException.ThrowIfNull(exception);
```

Comment:
NOTE: Exceptions thrown during the exception pipeline are currently **not** filtered out. If we want to filter these that should be done as a separate PR.